src/legacy/ui/public/agg_types/param_types/field.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { i18n } from '@kbn/i18n';
+import { AggConfig } from '../../vis';
+// @ts-ignore
+import { SavedObjectNotFound } from '../../errors';
+import { FieldParamEditor } from '../controls/field';
+// @ts-ignore
+import { BaseParamType } from './base';
+import { toastNotifications } from '../../notify';
+import { propFilter } from '../filter';
+import { Field } from '../../index_patterns';
+
+const filterByType = propFilter('type');
+
+export class FieldParamType extends BaseParamType {
+  editorComponent = FieldParamEditor;
+  required = true;
+  scriptable = true;
+  filterFieldTypes: string;
+  onlyAggregatable: boolean;
+
+  constructor(config: Record<string, any>) {
+    super(config);
+
+    this.filterFieldTypes = config.filterFieldTypes || '*';
+    this.onlyAggregatable = config.onlyAggregatable !== false;
+
+    if (!config.write) {
+      this.write = (aggConfig: AggConfig, output: Record<string, any>) => {
+        const field = aggConfig.getField();
+
+        if (!field) {
+          throw new TypeError(
+            i18n.translate(
+              'common.ui.aggTypes.paramTypes.field.requiredFieldParameterErrorMessage',
+              {
+                defaultMessage: '{fieldParameter} is a required parameter',
+                values: {
+                  fieldParameter: '"field"',
+                },
+              }
+            )
+          );
+        }
+
+        if (field.scripted) {
+          output.params.script = {
+            source: field.script,
+            lang: field.lang,
+          };
+        } else {
+          output.params.field = field.name;
+        }
+      };
+    }
+
+    this.serialize = (field: Field) => {
+      return field.name;
+    };
+
+    this.deserialize = (fieldName: string, aggConfig: AggConfig) => {
+      // @ts-ignore
+      const field = aggConfig.getIndexPattern().fields.byName[fieldName];
```

Comment:
Instead of `ts-ignore`-ing this completely, would something like this work?
```suggestion
      const field = aggConfig.getIndexPattern().fields.byName[fieldName] as Field;
```

Reply:
this tsignore was unnecessary it seems